core/field.ts

```diff
@@ -957,49 +998,47 @@ export abstract class Field<T = any> implements IASTNodeLocationSvg,
    * @param newValue New value.
    * @sealed
    */
-  setValue(newValue: AnyDuringMigration) {
+  setValue(newValue?: U|null) {
     const doLogging = false;
     if (newValue === null) {
       doLogging && console.log('null, return');
       // Not a valid value to check.
       return;
     }
 
-    let validatedValue = this.doClassValidation_(newValue);
-    // Class validators might accidentally forget to return, we'll ignore that.
-    newValue = this.processValidation_(newValue, validatedValue);
-    if (newValue instanceof Error) {
+    const classValidation = this.doClassValidation_(newValue);
+    const classValue = this.processValidation_(newValue, classValidation);
+    if (classValue instanceof Error) {
       doLogging && console.log('invalid class validation, return');
       return;
     }
+    // NOTE: newValue = undefined -> classValidation = null -> Error.
+    const newClassValue = classValue!;
```

Comment:
Although `processValidation` shouldn't return `null`, it could return `undefined` since `doClassValidation_` can be overwritten.

In our implementation, if `newValue = undefined`, `classValidation = null` so `classValue = Error` and returns out.

Theoretically, a subclass could implement `doClassValidation_` such that if `newValue = undefined`, `classValidation = undefined` which tells `processValidation_` to take `newValue`. This makes `classValue = newValue = undefined`. If that happens, it's *probably* a coding error by the subclass implementer, though there's a slim edge case where `T` includes `undefined`. (For example, a tri-state boolean checkbox might be `true|false|undefined` so `T=boolean|undefined`.)

As far as naming goes, I'm open to any suggestions. I originally named it `newClassValue` as just a mash up between `newValue` and `classValue`, not actually as a statement of it being non-null.